components/misc/misc.h

```diff
@@ -24,6 +24,65 @@ namespace Misc
       Point operator- (const Point &v) const { return {x - v.x, y - v.y}; }
       Point operator* (double c) const { return {static_cast<int32_t> (x * c), static_cast<int32_t> (y * c)}; }
     };
+
+    namespace detail
```

Comment:
Could you explain what all this is for? And maybe add a comment with explanation?

Reply:
There is a little comment at `overload` function. Basically this is the function which makes usage of `variant` classes bearable. If you have it you can write something like:
`overload ([](int){}, [](std::string){})` and use it as a variant visitor with additional benefits like lambda captures etc. Writing separate class for visitor is often cumbersome.